exonum-java-binding/testkit/src/test/java/com.exonum.binding.testkit/TestKitTest.java

```diff
@@ -0,0 +1,196 @@
+  }
+
+  @Test
+  void createTestKitMoreThanMaxServiceNumber() {
+    Class<IllegalArgumentException> exceptionType = IllegalArgumentException.class;
+    List<Class<? extends ServiceModule>> serviceModules = new ArrayList<>();
+    for (int i = 0; i < TestKit.MAX_SERVICE_NUMBER + 1; i++) {
+      serviceModules.add(TestServiceModule.class);
+    }
+    TestKit.Builder testKitBuilder = TestKit.builder(EmulatedNodeType.VALIDATOR)
+        .withServices(serviceModules);
+    assertThrows(exceptionType, testKitBuilder::build);
+  }
+
+  @Test
+  void createTestKitWithoutServices() {
+    Class<IllegalArgumentException> exceptionType = IllegalArgumentException.class;
+    TestKit.Builder testKitBuilder = TestKit.builder(EmulatedNodeType.VALIDATOR);
+    assertThrows(exceptionType, testKitBuilder::build);
+  }
+
+  public static final class TestServiceModule extends AbstractServiceModule {
+
+    private static final TransactionConverter THROWING_TX_CONVERTER = (tx) -> {
+      throw new IllegalStateException("No transactions in this service: " + tx);
+    };
+
+    @Override
+    protected void configure() {
+      bind(Service.class).to(TestService.class).in(Singleton.class);
+      bind(TransactionConverter.class).toInstance(THROWING_TX_CONVERTER);
+    }
+  }
+
+  public static final class TestServiceModule2 extends AbstractServiceModule {
+
+    private static final TransactionConverter THROWING_TX_CONVERTER = (tx) -> {
+      throw new IllegalStateException("No transactions in this service: " + tx);
+    };
+
+    @Override
+    protected void configure() {
+      bind(Service.class).to(TestService2.class).in(Singleton.class);
+      bind(TransactionConverter.class).toInstance(THROWING_TX_CONVERTER);
+    }
+  }
+
+  static final class TestService implements Service {
+
+    static short SERVICE_ID = 42;
+    static String SERVICE_NAME = "Test service";
+
+    @Override
+    public short getId() {
+      return SERVICE_ID;
+    }
+
+    @Override
+    public String getName() {
+      return SERVICE_NAME;
+    }
+
+    @Override
+    public Transaction convertToTransaction(RawTransaction rawTransaction) {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void createPublicApiHandlers(Node node, Router router) {
+      throw new UnsupportedOperationException();
+    }
+  }
+
+  static final class TestService2 implements Service {
+
+    static short SERVICE_ID = 48;
+    static String SERVICE_NAME = "Test service 2";
+
+    @Override
+    public short getId() {
+      return SERVICE_ID;
+    }
+
+    @Override
+    public String getName() {
+      return SERVICE_NAME;
+    }
+
+    @Override
+    public Transaction convertToTransaction(RawTransaction rawTransaction) {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void createPublicApiHandlers(Node node, Router router) {
+      throw new UnsupportedOperationException();
```

Comment:
We always call createPublicApiHandlers, even if no handlers are needed. I wonder if constructing this service will lead to panic